internal/testutils/fdtrace/fd.go

```diff
@@ -0,0 +1,38 @@
+package fdtrace
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"runtime"
+	"testing"
+
+	"github.com/cilium/ebpf/internal/sys"
+)
+
+// TestMain runs m with sys.FD leak tracing enabled.
+func TestMain(m *testing.M) {
+	sys.OnLeakFD(func(fs *runtime.Frames) {
```

Comment:
Second part of my suggestion:

```suggestion
	var ret int
	sys.OnLeakFD(func(fs *runtime.Frames) {
		ret = 42
		// ...
	})
	
	ret = m.Run()
	sys.OnLeakFD(nil)
	os.Exit(ret)
```

Reply:
Brilliant!